File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-0.cpp
class Solution {
   public:
    int characterReplacement(string s, int k) {
        int st = 0;
        int en = 0;
        int n = s.size();
        int maxi_freq = INT_MIN;
        unordered_map<char, int> mp;
        int ans = 0;

        while (en < n) {
            mp[s[en]]++;

            maxi_freq = max(maxi_freq, mp[s[en]]);

            if ((en - st + 1) - maxi_freq > k) {
                mp[s[st]]--;
                st++;
            }
            
            ans = max(ans, en - st + 1);
            en++;

        }
        return ans;
    }
};

File: Data Structures & Algorithms/balanced-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    int depth(TreeNode root){

        if(root == null)return 0;
        int ans = 0;
        ans = Math.max(ans , 1 + depth(root.left));
        ans = Math.max(ans , 1 + depth(root.right));
        return ans;
    }
    boolean solve(TreeNode root){
        if(root == null)return true;

        int l = depth(root.left);
        int r = depth(root.right);

        if(Math.abs(l - r) > 1)return false;

        return solve(root.left) && solve(root.right);
        

    }
    public boolean isBalanced(TreeNode root) {
        return solve(root);
    }
}
